NetworkingServiceKit/Classes/Networking/NetworkManager.swift

```diff
@@ -9,131 +9,184 @@
-        switch self {
-        case .tokenExpired( _):
-            return true
-        default:
-            return false
+        else {
+            let description: String
+            switch type {
+            case .persistenceValidation(let reason):
+                description = reason.description
+            case .responseValidation(let reason):
+                description = reason.description
+            }
+            self.details = MSErrorDetails(message: description, body: nil, path: nil, code: 0, underlyingError: nil)
         }
     }
-    init(code: Int, error: Error) {
+}
+
+// Mapped Error response failures
+public extension MSErrorType.ResponseFailureReason {
+
+    /// Conveience initializer
+    ///
+    /// - Parameter code: response code of the error
+    public init(code: Int) {
         switch code {
         case 401:
-            self = .tokenExpired(code: code)
+            self = .tokenExpired
+        case 403:
+            self = .forbidden
+        case 404:
+            self = .notFound
         case 500:
             self = .internalServerError
 
         default:
-            self = .badRequest(error: error)
+            self = code.isNetworkErrorCode ? .connectivity : .badRequest
         }
     }
 }
 
-extension MSError {
-    /// Returns whether the MSError is because our token expired
+public extension MSError {
+    /// Shortcut for identifying token expiration errors
     public var hasTokenExpired: Bool {
-        switch self {
-        case .responseValidationFailed(let reason):
-            return reason.hasTokenExpired
-        default: return false
-        }
-    }
-
-    //Returns the response code for an error
-    public var responseCode: Int {
-        switch self {
-        case .responseValidationFailed(let reason):
-            return reason.responseCode
+        switch type {
+        case .responseValidation(let reason):
+            return reason == .tokenExpired
         default:
-            return 0
+            return false
         }
     }
-
-    /// The `Error` returned by a system framework associated with a `.responseValidationFailed`
-    public var underlyingError: Error? {
-        switch self {
-        case .responseValidationFailed(let reason):
-            return reason.underlyingError
+    
+    /// Shortcut for identifying connectivity errors
+    public var isNetworkError: Bool {
+        switch type {
+        case .responseValidation(let reason):
+            return reason == .connectivity
         default:
-            return nil
+            return false
         }
     }
-
-    /// Helper method for determining whether an error is a connectivity issue
-    ///
-    /// - returns: returns true if the error is a connectivity issue and false if not
-    public var isNetworkError: Bool {
-        return responseCode.isNetworkErrorCode
+    
+    /// Returns a generic error to describe Core Data problems
+    static var genericLocalDataError: MSError {
```

Comment:
Seems more intuitive to mention that is a persistence error in the name like `genericPersistenceError`